File: engine/engine-model/src/main/scala/tools/packet/foreigneffect/CancelSkill.scala
package tools.packet.foreigneffect

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class CancelSkill(private var _fromCharacterId: Int, private var _skillId: Int) extends PacketInput {
  def fromCharacterId: Int = _fromCharacterId

  def skillId: Int = _skillId

  override def opcode(): SendOpcode = SendOpcode.CANCEL_SKILL_EFFECT
}

File: engine/engine-model/src/main/java/tools/packet/foreigneffect/CancelSkill.java
package tools.packet.foreigneffect;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record CancelSkill(Integer fromCharacterId, Integer skillId) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.CANCEL_SKILL_EFFECT;
   }
}